.github/workflows/unit-tests.yaml

```diff
@@ -9,44 +9,48 @@ on:
         description: 'The snap store channel to use to install the go snap'
         required: true
         type: string
-      unit-scenario:
-        description: 'The name of the scenario being tested: {normal, snapd-debug, withbootassetstesting, nosecboot, faultinject, race, snapdusergo}'
-        required: true
+      skip-coverage:
+        description: 'If true, will not generate test coverage files'
+        type: boolean
+        default: false
+        required: false
+      go-build-tags:
+        description: 'Tag to add to go test'
         type: string
+        required: false
+      go-test-race:
+        description: 'If true, will add race tag to go test'
+        type: boolean
+        default: false
+        required: false
+      snapd-debug:
+        description: 'If true, will set SNAPD_DEBUG=1'
+        type: boolean
+        default: false
+        required: false
 
 jobs:
   unit-tests:
     runs-on: ${{ inputs.runs-on }}
     env:
-      GOPATH: ${{ github.workspace }}
       # Set PATH to ignore the load of magic binaries from /usr/local/bin And
       # to use the go snap automatically. Note that we install go from the
       # snap in a step below. Without this we get the GitHub-controlled latest
       # version of go.
       PATH: /snap/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:${{ github.workspace }}/bin
       GOROOT: ""
+      GO_BUILD_TAGS: ${{ inputs.go-build-tags }}
+      SKIP_DIRTY_CHECK: 1
```

Comment:
Hmm, it is unclear why this is needed can you add a comment to the code please.

Reply:
It looks like it's actually not necessary. I just removed it.